packages/core/src/execution-engine/__tests__/execution-lifecycle-hooks.test.ts

```diff
@@ -0,0 +1,107 @@
+import { mock } from 'jest-mock-extended';
+import type {
+	IDataObject,
+	IExecuteResponsePromiseData,
+	INode,
+	IRun,
+	IRunExecutionData,
+	ITaskData,
+	IWorkflowBase,
+	Workflow,
+} from 'n8n-workflow';
+
+import type {
+	ExecutionLifecycleHookName,
+	ExecutionLifecyleHookHandlers,
+} from '../execution-lifecycle-hooks';
+import { ExecutionLifecycleHooks } from '../execution-lifecycle-hooks';
+
+describe('ExecutionLifecycleHooks', () => {
+	const executionId = '123';
+	const workflowData = mock<IWorkflowBase>();
+
+	let hooks: ExecutionLifecycleHooks;
+	beforeEach(() => {
+		jest.clearAllMocks();
+		hooks = new ExecutionLifecycleHooks('internal', executionId, workflowData);
+	});
+
+	describe('constructor()', () => {
+		it('should initialize with correct properties', () => {
+			expect(hooks.mode).toBe('internal');
+			expect(hooks.executionId).toBe(executionId);
+			expect(hooks.workflowData).toBe(workflowData);
+			expect(hooks.handlers).toEqual({
+				nodeExecuteAfter: [],
+				nodeExecuteBefore: [],
+				nodeFetchedData: [],
+				sendResponse: [],
+				workflowExecuteAfter: [],
+				workflowExecuteBefore: [],
+			});
+		});
+	});
+
+	describe('addHandler()', () => {
+		const hooksHandler =
+			mock<{
+				[K in keyof ExecutionLifecyleHookHandlers]: ExecutionLifecyleHookHandlers[K][number];
+			}>();
```

Comment:
Or `mock<ExecutionLifecyleHookHandlers>()`?

Reply:
couldn't get this to work